s2c_sensor_module/src/main.c

```diff
@@ -49,6 +53,9 @@ uint8_t board_id = 255;
 enum s2c_board_type board_type = S2C_BOARD_OTHER;
 struct s2c_board_config board_config;
 
+bool FLASH_BOARD_ID = true;
```

Comment:
In a resource constrained embedded system, every byte of flash or RAM is valuable. When statically configuring a behaviour, such as whether to flash the board ID or not, it's best to selectively include the code responsible for that feature, or to just not compile it in. You can do that with preprocessor macros. So:

`#define FLASH_BOARD_ID TRUE`

And then later, instead of having an if statement around the code that implements this feature, you would have a conditional macro:

```
#if (FLASH_BOARD_ID)
[code to flash board ID]
#endif
```

Now, if you need to turn this feature off, you simply set the macro to `FALSE`, and the whole code surrounded by the conditional macro will not get compiled in. This way, you don't waste any resources on unused functionality. Note that depending on how your compiler is configured, it might be smart enough to filter out the code the way you have it set up here, but using macros ensures that it's included or not with certainty.

Obviously the firmware in this repo is not at all at the flash or RAM limit of the microcontroller, but it's good to have this mentality when working with embedded systems at any scale, because as your code becomes more and more complex, you will be sure that you're using your limited resources wisely.